### DISCUSSION TITLE
ClashDetectionApis

### DISCUSSION THREAD

**Discussion by paprophet** [iTwin/itwinjs-core#1082] [Q&A]
in Clash Review Demo ,i want know how get the Clash elements by  local bim file  ? ecsql ? 
 because my file is change to bim file by snapshot and store in local .

**Comment by wilmaier** [comment]
I'm not sure I understand but if you have run clash detection for an iTwin iModel and want to review the clash results in a snapshot of the iModel it should work just like the Clash Review demo code.  You could still make the same API request to retrieve the clash results and then using the element id pairs you could call a function like visualizeClash() in ClashReviewApp.tsx:

https://www.itwinjs.org/sample-showcase/?group=iTwin+Platform+APIs&sample=clash-review-sample&imodel=Bay+Town+Process+Plant
